examples/counter/android/src/main/java/org/reduxkotlin/example/counter/MainActivity.kt

```diff
@@ -0,0 +1,49 @@
+package org.reduxkotlin.example.counter
+
+import android.os.Bundle
+import android.os.Handler
+import androidx.appcompat.app.AppCompatActivity
+import kotlinx.android.synthetic.main.activity_main.*
+import org.reduxkotlin.StoreSubscription
+import org.reduxkotlin.createStore
+import org.reduxkotlin.examples.counter.Decrement
+import org.reduxkotlin.examples.counter.Increment
+import org.reduxkotlin.examples.counter.reducer
+
+/**
+ * This is a sample of basic redux behavior.
+ * This is NOT best practice for structuring a multiplatform App.
+ */
+
+
+val store = createStore(reducer, 0)
+
+class MainActivity: AppCompatActivity() {
+    lateinit var storeSubscription: StoreSubscription
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        storeSubscription = store.subscribe { render(store.state) }
+        btn_increment.setOnClickListener { store.dispatch(Increment()) }
+        btn_decrement.setOnClickListener { store.dispatch(Decrement()) }
+        btn_async.setOnClickListener { incrementAsync() }
+        btn_increment_if_odd.setOnClickListener { incrementIfOdd() }
+    }
+
+    private fun render(state: Int) {
+        txt_label.text = "Clicked: $state times"
+    }
+
+    private fun incrementIfOdd() {
+        if (store.state % 2 != 0) {
```

Comment:
Should this kind of logic go in the reducer?

Reply:
again not best practice here, just going based on JS example.  In a MP project, def don't do this.  it is putting logic into the platform/view.  This is one pattern of creating an action.   This is also a really contrived example.  One way of doing this would be have a separate action named `class IncrementIfOdd` and do the logic in the reducer.  keep in mind the reducer can not dispatch actions.